packages/slate-commons/src/commands/isCursorInEmptyParagraph.ts

```diff
@@ -2,17 +2,13 @@ import type { Editor } from 'slate';
 import { Range } from 'slate';
 
 import { getCurrentNodeEntry } from './getCurrentNodeEntry';
-import type { IsEmptyParagraphElementOptions } from './isEmptyParagraphElement';
 import { isEmptyParagraphElement } from './isEmptyParagraphElement';
 
-export interface IsCursorInEmptyParagraphOptions {
-    shouldTrim?: IsEmptyParagraphElementOptions['shouldTrim'];
+interface Options {
```

Comment:
I think it is worth to export types that are used in exported functions because they are tight together. 
There can be difficulties when the consumer does not have access to the types that are used in exported entities

Reply:
>  There can be difficulties when the consumer does not have access to the types that are used in exported entities

Let's export them *when* we face such difficulties :)

Reply:
In the worst case you can always refer to function arguments via Typescript `Parameters` type:

```
type Options = Parameters<isEmptyParagraphElement>[2]; 
```

Reply:
@e1himself we could not face it, but the consumers could, like here https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57710

Also there can be potential issues with naming as Options pretty general name and when you will need to export it probably you would need to rename the interface as well.

Reply:
@e1himself yes, but it will not work with generics

Reply:
> Also there can be potential issues with naming as Options pretty general name and when you will need to export it probably you would need to rename the interface as well.

IDK, I see what you mean. But I hate making the code difficult to read without having a really good reason. The ones you named I classify as "potential" -- meaning some people may face it in the future, or not. So I'd rather keep the code simpler until there's a real reason to complicate it.